internal/store/store.go

```diff
@@ -89,235 +106,268 @@ func customCATransport(ca []byte) http.RoundTripper {
+// Service configures the service.
+func Service(service string) Option {
+	return func(s *Store) error {
+		s.service = service
+		return nil
 	}
-
-	return s, err
 }
 
-// NewUnauthenticated a client to interact with the Ops Manager APIs that don't require authentication
-func NewUnauthenticated(c Config) (*Store, error) {
-	s := new(Store)
-	s.service = c.Service()
-
-	if s.service != config.OpsManagerService {
-		return nil, fmt.Errorf("unsupported service: %s", s.service)
+// WithBaseURL configures the base URL for the underling client.
+// the url should not contain any path, to add the public API path use WithPublicPathBaseURL.
+func WithBaseURL(configURL string) Option {
+	return func(s *Store) error {
+		s.baseURL = configURL
+		return nil
 	}
-	if configURL := c.OpsManagerURL(); configURL != "" {
-		s.baseURL = s.apiPath(configURL)
+}
+
+// WithPublicPathBaseURL if you use WithBaseURL and need the Store to connect to the public API.
+func WithPublicPathBaseURL() Option {
+	return func(s *Store) error {
+		if s.service == config.CloudService {
+			s.baseURL += atlas.APIPublicV1Path
+			return nil
+		}
+		s.baseURL += opsmngr.APIPublicV1Path
+		return nil
 	}
-	if caCertificate := c.OpsManagerCACertificate(); caCertificate != "" {
+}
+
+// WithCACertificate when using a custom CA certificate.
+func WithCACertificate(caCertificate string) Option {
+	return func(s *Store) error {
 		s.caCertificate = caCertificate
+		return nil
 	}
-	if skipVerify := c.OpsManagerSkipVerify(); skipVerify != yes {
-		s.skipVerify = skipVerify
-	}
-	client, err := defaultClient(c)
-	if err != nil {
-		return nil, err
-	}
-	if err := s.setOpsManagerClient(client); err != nil {
-		return nil, err
-	}
-	return s, nil
 }
 
-// NewVersionManifest ets the appropriate client for the manifest version page
-func NewVersionManifest(c Config) (*Store, error) {
-	s := new(Store)
-	s.service = c.Service()
-	if s.service != config.OpsManagerService {
-		return nil, fmt.Errorf("unsupported service: %s", s.service)
+// SkipVerify skips CA certificate verification, use at your own risk.
+func SkipVerify() Option {
+	return func(s *Store) error {
+		s.skipVerify = true
+		return nil
 	}
-	s.baseURL = versionManifestStaticPath
+}
 
-	if baseURL := c.OpsManagerVersionManifestURL(); baseURL != "" {
-		s.baseURL = baseURL
+// CredentialsGetter how to get credentials when Store must be authenticated.
+type CredentialsGetter interface {
+	PublicAPIKey() string
+	PrivateAPIKey() string
+}
+
+// WithAuthentication sets the store credentials
+func WithAuthentication(c CredentialsGetter) Option {
+	return func(s *Store) error {
+		s.username = c.PublicAPIKey()
+		s.password = c.PrivateAPIKey()
+		return nil
 	}
+}
 
-	client, err := defaultClient(c)
-	if err != nil {
-		return nil, err
+func withAtlasClient(client *http.Client) Option {
+	return func(s *Store) error {
+		opts := []atlas.ClientOpt{atlas.SetUserAgent(userAgent)}
```

Comment:
[q] why we need `userAgent`?

Reply:
why do we need the variable or why do we set it?

the variable is there so we generate it once instead of every time we create a client, also it's shared with the ops manager client

Reply:
sorry If I wasn’t clear, I am not sure what the userAgent is used for. Why do we need to set a userAgent? and what the userAgent does?

Reply:
So Atlas knows who/what made the HTTP call, https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent